code.js

```diff
@@ -1,40 +1,42 @@
+ 
 // References:
 // https://www.geeksforgeeks.org/binary-search-in-javascript/ for help
 // https://github.com/COSC3020/binary-search-ZachRenz/blob/main/code.js
 // https://www.youtube.com/watch?v=P3YID7liBug (helped for visualization of algorithm)
-
+ 
 function binarySearch(list, element) {
-    let mid;
     if(list.length == 1){
         if(list[0] == element){
             return 0;
-        }
-    }
-    if(list.length % 2 == 0){
-        mid = Math.floor((list.length-1)/2);
-    } else {
-        mid = Math.floor((list.length)/2);
-    }
-
-    
-    if(list[mid] == element){
-        let lower = binarySearch(list.slice(0, mid), element);
-        if(lower == -1){
-            return mid;
         } else {
-            return lower;
+            return -1;
         }
-    } else if(element < list[mid]) {  
-        if(mid == 1){
-            if(list[0] == element){
-                return 0;
+    } else {
+        let mid = Math.floor((list.length-1)/2);
+        if(list[mid] == element){
+            if(binarySearch(list.slice(0, mid), element) == -1){
+                return mid;
+            } else {
+                return binarySearch(list.slice(0, mid), element);
+            }
+        } else if(element < list[mid]) { 
+            return binarySearch(list.slice(0, mid), element);
+        } else if(element > list[mid]){
+            if(binarySearch(list.slice(mid+1, list.length), element) != -1){
+                return ++mid + binarySearch(list.slice(mid, list.length), element);
             } else {
                 return -1;
             }
-        }      
-        return binarySearch(list.slice(0, mid-1), element);
-    } else if(element > list[mid]){
-        return mid+1 + binarySearch(list.slice(mid+1, list.length), element);
+        }
+        return -1;
     }
-    return -1;
 }
+
```

Comment:
Thanks for the testing part, but it's better not to be in the main code.

Reply:
Thanks for the feedback! Do you have any suggestions on where I should put the testing section? Would it be helpful to have another file that tests the code for specific cases or would it just be better to remove the test section?

Reply:
You can put it in a separate file or even remove it completely, whatever suits you! The reason for test cases is to check the answer for yourself during the coding.